### DISCUSSION TITLE
Example config question

### DISCUSSION THREAD

**Discussion by x-Felix** [madnuttah/unbound-docker#79] [Q&A] [answered]
I am taking this [example config](https://raw.githubusercontent.com/madnuttah/unbound-docker/refs/heads/main/doc/examples/docker-compose-madnuttah.yaml) as reference to add Redis cache for unbound. 
Curious of the docket networks assignment in the docker compose config.

`pihole` container has two networks (`docker`, `docker-bridge`) attached. 
`unbound` container has only `docker` network attached. 
`unbound-db-socket` and `unbound-db` are all in `docker-bridge` network. 

Suppose pihole only needs to talk to unbound, why does it need docker-bridge network? 
unbound suppose having dependence on unbound-db-socket and unbound-db, but they are in different networks, how they communicate? 

I haven't tested the example config in my home production environment. Just want to make sure I fully understand the settings before pulling the trigger. 

Thanks.

**Comment by madnuttah** [comment]
Hey @x-Felix and thanks for reaching out!

That's pretty much the compose file I'm using. I prefer a `macvlan` configuration and the issue is that the host can't communicate with the pi-hole container (and vice versa, of course), which is needed for the host's dns lookups in that case. To circumvent this, a separate bridge network was attached to pi-hole. I don't allow access to unbound from any device but pi-hole, so a pure `macvlan` config is sufficient.

The networks for the unbound-db and unbound-db-socket ain't really needed.

In case you ask yourself, >>why `macvlan`<<? I have firewall rules.

Cheers!

Edit: I hope you don't mind, I have converted this into a discussion.

**Comment by x-Felix** [comment]
Thanks @madnuttah . I have no question of pihole and unbound using macvlan approach. But you are saying the unbound, unbound-db and unbound-db-socket instances does not need to be in the same docker network? How they communicate then?

**Reply by madnuttah** [reply, marked as answer]
Unbound needs network, the db and socket don't neccessarily need a network, they just serve redis or the socket volume respectively and are accessed by Unbound through the UNIX socket. If you'd like to use redis with TCP/IP instead of socket, a network for the unbound-db would be needed.

**Comment by x-Felix** [comment]
Thanks, @madnuttah. 
Can I ask another question? In the documentation https://github.com/madnuttah/unbound-docker/blob/main/doc/README.md#cachedb-redis section. It says "Create a new mountpoint like ../unbound-db/, make it available via fstab and place this [redis.conf](https://raw.githubusercontent.com/madnuttah/unbound-docker/main/doc/examples/redis/redis.conf) there." I cannot figure out the fstab part. Would you please elaborate it a bit more details? I just place redis.conf under ../unbound-db/ and map it in docker-compose.yaml volumes config. Thanks.

**Reply by madnuttah** [reply]
Yeah, this can be done better for sure:

- Remove `volumes_from` stuff
- Set `- cachedb.d:/usr/local/unbound/cachedb.d/` in volumes section of `unbound` and `unbound-db-socket`

Edit: I just verified it's working, many thanks for the heads-up! Do you want to fix that in example and documentation? :)

**Reply by x-Felix** [reply]
Sorry, I am still not 100% sure the fstab part. Does that mean need to have cachedb volume available on host as well? It will be good if you can update the document. Thanks.

**Reply by madnuttah** [reply]
Don't mind the fstab part, I'll adapt the documentation.

Thanks

**Reply by madnuttah** [reply]
It's done.
